src/storage/table/column_data.cpp

```diff
@@ -24,9 +24,11 @@ namespace duckdb {
 ColumnData::ColumnData(BlockManager &block_manager, DataTableInfo &info, idx_t column_index, idx_t start_row,
                        LogicalType type_p, optional_ptr<ColumnData> parent)
     : start(start_row), count(0), block_manager(block_manager), info(info), column_index(column_index),
-      type(std::move(type_p)), parent(parent), allocation_size(0) {
+      type(std::move(type_p)), allocation_size(0), parent(parent) {
 	if (!parent) {
 		stats = make_uniq<SegmentStatistics>(type);
+	} else if (type.id() == LogicalTypeId::VALIDITY) {
+		parent->validity = this;
```

Comment:
This is currently unused, in the future it'll be used to reference the validity ColumnData from the base data
Perhaps this is not entirely necessary though, it might be nicer to create a virtual method that `StandardColumnData` overrides, as the `validity` would mostly (only?) be relevant for that